src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaGeneratorDatabaseSchemaLoader.cs

```diff
@@ -26,6 +32,26 @@ public EntityStoreSchemaGeneratorDatabaseSchemaLoader(EntityConnection entityCon
 
             _connection = entityConnection;
             _storeSchemaModelVersion = storeSchemaModelVersion;
+
+            var switchOffMetadataMergeJoins = false;
+            var metadataMergeJoinsAppSettings =
+                ConfigurationManager.AppSettings.GetValues(SwitchOffMetadataMergeJoins);
+            if (metadataMergeJoinsAppSettings != null)
+            {
+                var metadataMergeJoinsAppSetting =
+                    metadataMergeJoinsAppSettings.FirstOrDefault();
+                if (metadataMergeJoinsAppSetting != null)
+                {
+                    bool.TryParse(
+                        metadataMergeJoinsAppSetting, out switchOffMetadataMergeJoins);
+                }
+            }
+            if (!switchOffMetadataMergeJoins
+                && _connection != null
+                && typeof(System.Data.SqlClient.SqlConnection) == _connection.StoreConnection.GetType())
```

Comment:
Not sure that this is the best way to test for SQL Server because when a wrapping provider is being used the connection type can be something different than SqlConnection. Is there something in the SSDL metadata that indicates SQL Server?

Reply:
Understood. Two approaches I can take here:
1. I can look for the string "provider=System.Data.SqlClient" in the existing `entityConnection.ConnectionString`.
2. I can pass in the `ModelBuilderSettings` object (which has all the information we've gathered about the provider, including e.g. the `RuntimeProviderInvariantName`) into the constructor of `EntityStoreSchemaGeneratorDatabaseSchemaLoader`.

Option 1 will probably need a regex - which we were trying to avoid - but we'd need to make sure we'd reached the end of the string or that the next character is a separator character.
Option 2 implies changing a public constructor but of an internal class - so not a breaking change.

I think I prefer option 2. Are you OK with that?

Reply:
Actually the 2nd approach causes `EntityDesignerVersioningFacade` to have a dependency on `EntityDesign` which it should not have. However there is a 3rd approach -  can just pass a `bool` saying whether we need to provide the MERGE JOIN hint. That way the caller (who has the information) can decide. I'll include that in the next commit.